pkg/ring/kv/memberlist/memberlist_client.go

```diff
@@ -632,38 +687,45 @@ func (m *Client) GetBroadcasts(overhead, limit int) [][]byte {
 // This is "pull" part of push/pull sync (either periodic, or when new node joins the cluster).
 // Here we dump our entire state -- all keys and their values. There is no limit on message size here,
 // as Memberlist uses 'stream' operations for transferring this state.
-func (m *Client) LocalState(join bool) []byte {
+func (m *KV) LocalState(join bool) []byte {
 	m.numberOfPulls.Inc()
 
 	m.storeMu.Lock()
 	defer m.storeMu.Unlock()
 
-	// For each Key/Value pair in our store, we write:
-	// [1 byte key length] [key] [4-bytes value length] [value]
+	// For each Key/Value pair in our store, we write
+	// [4-bytes length of KV pair] [KV pair]
```

Comment:
`length of KV pair` > `length of marshalled KV pair` may be more clear (I didn't get it at the beginning)